RMS/BufferedCapture.py

```diff
@@ -905,41 +958,169 @@ def releaseResources(self):
+            self.startup_frames = 25*60*10  # 10 minutes
+            self.b = 0
+            self.b_error_debt = 0
+            self.m_jump_error = 0
+            self.last_m_err = float('inf')
+            self.last_m_err_n = 0
+
+            # Initialize mode tracking
+            self.last_daytime_mode = self.daytime_mode.value
+            self.current_raw_frame_shape = None
+
+            # Initialize raw frame handling if enabled
+            if self.config.save_frames:
+                self.raw_frame_count = 0
+                
+                # Convert shared timestamp arrays to numpy arrays
+                self.sharedTimestamps = np.ctypeslib.as_array(self.sharedTimestampsBase.get_obj())
+                self.sharedTimestamps2 = np.ctypeslib.as_array(self.sharedTimestampsBase2.get_obj())
+
+                # Raw frame arrays will be initialized after we know the frame shape
+                self.sharedRawArrayBase = None
+                self.sharedRawArray = None
+                self.sharedRawArrayBase2 = None
+                self.sharedRawArray2 = None
+                self.raw_frame_saver = None
+
+            log.debug("Process-specific initialization complete")
+
+            # Main capture loop
+            while not self.exit.is_set() and not self.initVideoDevice():
+                log.info('Waiting for the video device to be connected...')
+                time.sleep(5)
+
+            if self.device is None:
+                log.info('The video source could not be opened!')
+                self.exit.set()
+                return False
+
+            # If we get here, we have a valid device and frame_shape
+            # Initialize raw frame arrays and saver if enabled
+            if self.config.save_frames and self.frame_shape is not None:
+                try:
+                    log.debug("Initializing raw frame arrays with shape: {}".format(self.frame_shape))
+
+                    self.initRawFrameArrays(self.frame_shape)
+
+                    # Initialize and start frame saver
+                    log.debug("Creating and starting RawFrameSaver...")
+                    self.raw_frame_saver = RawFrameSaver(self.saved_frames_dir, 
+                                                    self.sharedRawArray, self.startRawTime1, 
+                                                    self.sharedRawArray2, self.startRawTime2, 
+                                                    self.sharedTimestamps, self.sharedTimestamps2,
+                                                    self.config)
+                    self.raw_frame_saver.start()
+                    log.debug("Raw frame handling initialization complete")
+                    
+                except Exception as e:
+                    log.error("Failed to initialize raw frame handling: {}".format(e))
+                    log.debug(repr(traceback.format_exception(*sys.exc_info())))
+                    self.raw_frame_saver = None
+                    raise

```

Comment:
I think we can safely delete L1087-L1107 since we are initializing the arrays on frame shape change later. Also the frame_shape at this point is the code is not accurate yet as it's not gone through the greyscale conversion if applicable.. 
Later at L1264,  `if frame.shape != self.current_raw_frame_shape:` is the test to (re)initialize the shared arrays. At that point frame.shape should be correct (as it's been read() ) and current_raw_frame_shape is None before the first initialization.